internal/sqladapter/store_test.go

```diff
@@ -0,0 +1,209 @@
+package sqladapter_test
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/pressly/goose/v3"
+	"github.com/pressly/goose/v3/internal/check"
+	"github.com/pressly/goose/v3/internal/sqladapter"
+	"github.com/pressly/goose/v3/internal/testdb"
+	"go.uber.org/multierr"
+	"modernc.org/sqlite"
+	_ "modernc.org/sqlite"
+)
+
+// The goal of this test is to verify the sqladapter package works as expected. This test is not
+// meant to be exhaustive or test every possible database dialect. It is meant to verify the Store
+// interface works against a real database.
+
+func TestStore(t *testing.T) {
```

Comment:
This allows us to test the Store implementation in isolation from migrations.

```
┌───────────────────────────────────────────────────────────────────────────────────────────────────┐
│  STATUS │ ELAPSED │                     PACKAGE                     │ COVER │ PASS │ FAIL │ SKIP  │
│─────────┼─────────┼─────────────────────────────────────────────────┼───────┼──────┼──────┼───────│
│  PASS   │  1.94s  │ github.com/pressly/goose/v3/internal/sqladapter │ 70.8% │  3   │  0   │  0    │
└───────────────────────────────────────────────────────────────────────────────────────────────────┘
```

There's both a short (sqlite3 with a temp test directory) and long (postgres container) test